src/javascript/app/pages/trade/barriers.js

```diff
@@ -160,14 +161,73 @@ const Barriers = (() => {
         Defaults.remove('barrier', 'barrier_high', 'barrier_low');
     };
 
-    const validateBarrier = () => {
+    const showError = (barrier) => {
+        if (!barrier.classList.contains('error-field')) {
+            barrier.classList.add('error-field');
+        }
+        const error_node = barrier.parentNode.getElementsByClassName('hint')[0].getElementsByClassName('error-msg')[0];
+        if (error_node.classList.contains('invisible')) {
+            error_node.classList.remove('invisible');
+        }
+    };
+
+    const hideError = (barrier) => {
+        if (barrier.classList.contains('error-field')) {
+            barrier.classList.remove('error-field');
+        }
+        const error_node = barrier.parentNode.getElementsByClassName('hint')[0].getElementsByClassName('error-msg')[0];
+        if (!error_node.classList.contains('invisible')) {
+            error_node.classList.add('invisible');
+        }
+    };
+
+    /**
+    * Show an error on the target Barrier.
+    *
+    * @param {Object} barrier_1   the target barrier object for prompting error
+    * @param {Object} barrier_2   barrier object whose errors will be resolved
+    */
+    const showErrorOnTarget = (barrier_1, barrier_2) => {
+        showError(barrier_1);
+        hideError(barrier_2);
+    };
+
+    const resolveAllErrors = (barrier_1, barrier_2) => {
+        hideError(barrier_1);
+        hideError(barrier_2);
+    };
+
+    /**
+    * Validate Barriers
+    * @param {Boolean} is_high_barrier_changed Whether we're validating this barrier.
+    *                                          And the default validation is on High barrier.
+    *
+    */
+    const validateBarrier = (is_high_barrier_changed = true) => {
         const barrier_element = getElementById('barrier');
         const empty           = isNaN(parseFloat(barrier_element.value)) || parseFloat(barrier_element.value) === 0;
+        const barrier_high_element = getElementById('barrier_high');
+        const barrier_low_element = getElementById('barrier_low');
+
         if (isVisible(barrier_element) && empty) {
             barrier_element.classList.add('error-field');
         } else {
             barrier_element.classList.remove('error-field');
         }
+
+        const is_high_barrier_greater = +barrier_high_element.value > +barrier_low_element.value;
+
+        if (is_high_barrier_changed) {
+            if (is_high_barrier_greater) {
+                resolveAllErrors(barrier_high_element, barrier_low_element);
+            } else {
+                showErrorOnTarget(barrier_high_element, barrier_low_element);
+            }
+        } else if (is_high_barrier_greater) {
+            resolveAllErrors(barrier_high_element, barrier_low_element);
+        } else {
+            showErrorOnTarget(barrier_low_element, barrier_high_element);
+        }
```

Comment:
how about simplifying this a bit? something like:

```
if (+barrier_high_element.value > +barrier_low_element.value) {
    hideError(barrier_high_element);
    hideError(barrier_low_element);
} else {
    hideError(is_high_barrier_changed ? barrier_low_element : barrier_high_element);
    showError(is_high_barrier_changed ? barrier_high_element : barrier_low_element);
}
```

Reply:
oh no... it causes PageSpeed loss only when we find for elements in `if` statement :) nvm.